modelmapper/cleaner.py

```diff
@@ -60,16 +72,35 @@ def get_csv_data_cleaned(self, path_or_content, original_content_type=None):
             try:
                 field_info = model_info[field_name]
             except KeyError:
-                raise KeyError(FIELD_NAME_NOT_FOUND_MSG.format(field_name)) from None
+                if ignore_new_fields:
+                    self._missing_fields.add(field_name)
+                    continue
+                else:
+                    raise KeyError(FIELD_NAME_NOT_FOUND_MSG.format(field_name)) from None
             self._get_field_values_cleaned_for_importing(field_name, field_info, field_values, original_content_type)
+        if ignore_new_fields and self._missing_fields:
+            for field in self._missing_fields:
+                all_items.pop(field)
+
+            if not self.publicized:
+                self.logger.error(
+                    """Fields in the report not found in Model. Model will need to be retrained.
+                    Fields missing: {}. Model: {}""".format(
+                        self._missing_fields, self.settings.combined_file_name[:-3]),
+                )
+
+                self.slack_missing_fields()
+                self.publicized = True
 
-        # transposing
         all_lines_cleaned = zip(*all_items.values())
 
         for i in all_lines_cleaned:
             yield dict(zip(all_items.keys(), i))
 
     def _get_field_values_cleaned_for_importing(self, field_name, field_info, field_values, original_content_type):
+        """ Reads in training values to generate models??
```

Comment:
I don't think this is for training. This is importing the data and preparing it for insertion into the database through type checking and conversion by getting the field type from the DB model and checking other properties on the toml file. For example if it's a nullable field it allows for nulls, if its an integer field it pulls a number out of the value, if its a percentage it divides that number by 100, etc...
So a more appropriate docstring would be (though I usually don't go into this much detail for an `_` method):
```python
def _get_field_values_cleaned_for_importing(self, field_name, field_info, field_values, original_content_type):
    """Prepares source data for insertion into database.

    Arguments:
        field_name (str) - The name of the field to be inserted into the model.
        field_info (dict) - Information about the filed pulled from the model's TOML file.
        field_values (list) - All values to be inserted into the model from the source data.
        original_content_type (str) - The file type of the source data (i.e. xls or csv)

    Returns:
        field_values (list) - The prepared values to insert into the given model.

    Raises:
        ValueError, TypeError - Indicates something is wrong with the incoming data, refer to error message.
    """

Reply:
👍